pkg/asset/agent/image/ignition.go

```diff
@@ -126,7 +126,8 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 
 	a.RendezvousIP = nodeZeroIP
 
-	// TODO: don't hard-code target arch
+	archName := manifests.NormalizeArch(infraEnv.Spec.CpuArchitecture)
```

Comment:
IIUC this is the one case where we need to 'normalize', due to assisted's previous practice of using the CoreOS arch string for `x86_64` and the Docker string for `arm64`?

I think in this case I think it would be best to use one of the new functions you added in https://github.com/openshift/assisted-service/pull/4441 to do the conversion.